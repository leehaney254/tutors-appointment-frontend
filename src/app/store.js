import { configureStore } from '@reduxjs/toolkit';
import registerReducer from '../features/RegisterSlice';
import loginReducer from '../features/LoginSlice';

export default configureStore({
  reducer: {
    register: registerReducer,
    login: loginReducer,
  },
});
