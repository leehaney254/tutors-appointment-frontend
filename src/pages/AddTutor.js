import React from "react";

const AddTutor = () => {
  return <div>Add tutor page</div>;
};

export default AddTutor;
