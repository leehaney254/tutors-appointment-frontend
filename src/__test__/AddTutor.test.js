import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../app/store';
import AddTutor from '../pages/AddTutor';

it('should render AddTutor page', () => {
  const addTutor = render(
    <BrowserRouter>
      <Provider store={store}>
        <AddTutor />
      </Provider>
    </BrowserRouter>,
  );
  expect(addTutor).toMatchSnapshot();
});
