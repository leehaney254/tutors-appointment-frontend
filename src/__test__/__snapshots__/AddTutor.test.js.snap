// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddTutor page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="text-center"
        >
          Add Tutor
        </h2>
        <hr
          class="border-2 w-full mb-3"
        />
        <form
          class="scale-x-75"
        >
          <div
            class="mb-6"
          >
            <label
              class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              for="name"
            >
              Name
            </label>
            <input
              class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
              id="name"
              name="name"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-6"
          >
            <label
              class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              for="image"
            >
              Image
            </label>
            <input
              class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
              id="Image"
              name="image"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-6"
          >
            <label
              class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              for="speciality"
            >
              Speciality
            </label>
            <input
              class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
              id="specaility"
              name="speciality"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-6"
          >
            <label
              class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              for="bio"
            >
              Bio
            </label>
            <textarea
              class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
              id="username"
              name="bio"
              required=""
              type="text"
            />
          </div>
          <div
            class="mb-6"
          >
            <label
              class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
              for="price"
            >
              Price
            </label>
            <input
              class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
              id="price"
              min="0"
              name="price"
              required=""
              type="number"
              value="0"
            />
          </div>
          <button
            class="text-white bg-green-700 hover:bg-green-800 focus:ring-4 focus:outline-none focus:ring-green-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-green-600 dark:hover:bg-green-700 dark:focus:ring-green-800"
            type="submit"
          >
            Create Tutor
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="text-center"
      >
        Add Tutor
      </h2>
      <hr
        class="border-2 w-full mb-3"
      />
      <form
        class="scale-x-75"
      >
        <div
          class="mb-6"
        >
          <label
            class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            for="name"
          >
            Name
          </label>
          <input
            class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
            id="name"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-6"
        >
          <label
            class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            for="image"
          >
            Image
          </label>
          <input
            class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
            id="Image"
            name="image"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-6"
        >
          <label
            class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            for="speciality"
          >
            Speciality
          </label>
          <input
            class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
            id="specaility"
            name="speciality"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-6"
        >
          <label
            class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            for="bio"
          >
            Bio
          </label>
          <textarea
            class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
            id="username"
            name="bio"
            required=""
            type="text"
          />
        </div>
        <div
          class="mb-6"
        >
          <label
            class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
            for="price"
          >
            Price
          </label>
          <input
            class="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 dark:shadow-sm-light"
            id="price"
            min="0"
            name="price"
            required=""
            type="number"
            value="0"
          />
        </div>
        <button
          class="text-white bg-green-700 hover:bg-green-800 focus:ring-4 focus:outline-none focus:ring-green-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-green-600 dark:hover:bg-green-700 dark:focus:ring-green-800"
          type="submit"
        >
          Create Tutor
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
